### ISSUE TITLE
/cluster/requestsizer/planV2 only generates a plan for one recommendation

### ISSUE THREAD

**Issue by gcsabbagh** [kubecost/cost-analyzer-helm-chart#2423]
**Describe the bug**  
The `planV2` API only generates a plan for the last recommendation passed in the recommendations array in the request body.

**To Reproduce**  

1. Call `/cluster/requestsizer/planV2` API with a request body that includes multiple recommendations
2. Inspect the response
3. Only one recommendation generated a plan (the last recommendation in the `Recommendations[]` array)

**Sample Body**
```
{
    "TotalMonthlySavings": 101.28881670898437,
    "Count": 78,
    "Recommendations": [
        {
            "clusterID": "cluster-one",
            "namespace": "kirollos-botros",
            "controllerKind": "deployment",
            "controllerName": "knobby-quokka-llc-core",
            "containerName": "appserver",
            "recommendedRequest": {
                "cpu": "50m",
                "memory": "1124Mi"
            },
            "monthlySavings": {
                "cpu": 10.388812500000004,
                "memory": 2.7922756640624993
            },
            "latestKnownRequest": {
                "cpu": "500m",
                "memory": "2Gi"
            },
            "currentEfficiency": {
                "cpu": 0.005473685687277871,
                "memory": 0.4268336356901009,
                "total": 0.15253898513043188
            }
        },
        {
            "clusterID": "cluster-one",
            "namespace": "kirollos-botros",
            "controllerKind": "deployment",
            "controllerName": "bald-aardwolf-llc-core",
            "containerName": "appserver",
            "recommendedRequest": {
                "cpu": "50m",
                "memory": "1291Mi"
            },
            "monthlySavings": {
                "cpu": 10.388812500000004,
                "memory": 2.2876111230468745
            },
            "latestKnownRequest": {
                "cpu": "500m",
                "memory": "2Gi"
            },
            "currentEfficiency": {
                "cpu": 0.005380044037717611,
                "memory": 0.5021185389010236,
                "total": 0.17875436566350167
            }
        }
    ]
}
```

**Expected behavior**  
The planV2 response should include a plan for each recommendation passed to it

**Slack thread**  
https://kubecost.slack.com/archives/CLFV60Y90/p1688502636556489?thread_ts=1688408989.047539&cid=CLFV60Y90

**What impact will this have on your ability to get value out of Kubecost?**

I need to write a wrapper script to plan each recommendation individually. This generates potentially hundreds of calls instead of a single call to `planV2`

**Bug report**
Please generate a bug report and attach the resulting `kubecost-bug-report.txt` here:
https://docs.kubecost.com/troubleshooting/bug-report

[kubecost-bug-report.txt](https://github.com/kubecost/cost-analyzer-helm-chart/files/11986412/kubecost-bug-report.txt)

**Comment by gcsabbagh** [comment]
FYI This bug happens on both `v1.103.5` and `v1.104.4` that I've tested so far

**Comment by kirbsauce** [comment]
@kwombach12 , @gcsabbagh filed this at the request of support.  User is working with our beta APIs.

**Comment by kwombach12** [comment]
Thanks @kirbsauce @gcsabbagh We will take a look at this!

**Comment by gcsabbagh** [comment]
Why did we close the issue? @nickcurie this bug still happens on v1.105.1

**Comment by kwombach12** [comment]
@gcsabbagh apologies for the pre-mature close - we have a fix ready to go that will be release in 1.106. I will leave the issue open until you confirm that the fix works for you.

**Comment by chipzoller** [comment]
Closing, stale.
